filters/auth/oidc_introspection.go

```diff
@@ -0,0 +1,167 @@
+package auth
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	"github.com/tidwall/gjson"
+
+	log "github.com/sirupsen/logrus"
+	"github.com/zalando/skipper/filters"
+)
+
+const (
+	OidcClaimsQueryName = "oidcClaimsQuery"
+	oidcClaimsCacheKey  = "oidcclaimscachekey"
+)
+
+type (
+	oidcIntrospectionSpec struct {
+		typ roleCheckType
+	}
+	oidcIntrospectionFilter struct {
+		typ   roleCheckType
+		paths []pathQuery
+	}
+
+	pathQuery struct {
+		path    string
+		queries []string
+	}
+)
+
+func NewOIDCQueryClaimsFilter() filters.Spec {
+	return &oidcIntrospectionSpec{
+		typ: checkOIDCQueryClaims,
+	}
+}
+
+func (spec *oidcIntrospectionSpec) Name() string {
+	switch spec.typ {
+	case checkOIDCQueryClaims:
+		return OidcClaimsQueryName
+	}
+	return AuthUnknown
+}
+
+func (spec *oidcIntrospectionSpec) CreateFilter(args []interface{}) (filters.Filter, error) {
+	sargs, err := getStrings(args)
+	if err != nil {
+		return nil, err
+	}
+	if len(sargs) == 0 {
+		return nil, filters.ErrInvalidFilterParameters
+	}
+
+	filter := &oidcIntrospectionFilter{typ: spec.typ}
+
+	switch filter.typ {
+	case checkOIDCQueryClaims:
+		for _, arg := range sargs {
+			slice := strings.SplitN(arg, ":", 2)
+			if len(slice) != 2 {
+				return nil, fmt.Errorf("%v: malformatted filter arg %s", filters.ErrInvalidFilterParameters, arg)
+			}
+			pq := pathQuery{path: slice[0]}
+			for _, query := range strings.Split(slice[1], " ") {
+				if query == "" {
+					return nil, fmt.Errorf("%v: %s", errUnsupportedClaimSpecified, arg)
+				}
+				pq.queries = append(pq.queries, trimQuotes(query))
+			}
+			if len(pq.queries) == 0 {
+				return nil, fmt.Errorf("%v: %s", errUnsupportedClaimSpecified, arg)
+			}
+			filter.paths = append(filter.paths, pq)
+		}
+		if len(filter.paths) == 0 {
+			return nil, fmt.Errorf("%v: no queries could be parsed", errUnsupportedClaimSpecified)
+		}
+	default:
+		return nil, filters.ErrInvalidFilterParameters
+	}
+
+	return filter, nil
+}
+
+func (filter *oidcIntrospectionFilter) String() string {
+	var str []string
+	for _, query := range filter.paths {
+		str = append(str, query.String())
+	}
+	return fmt.Sprintf("%s(%s)", OidcClaimsQueryName, strings.Join(str, "; "))
+}
+
+func (filter *oidcIntrospectionFilter) Request(ctx filters.FilterContext) {
+	r := ctx.Request()
+
+	token, ok := ctx.StateBag()[oidcClaimsCacheKey].(tokenContainer)
+	if !ok || &token == (&tokenContainer{}) || len(token.Claims) == 0 {
+		log.Errorf("Error retrieving oidcClaimsCacheKey for OIDC token introspection")
+		unauthorized(ctx, "", missingToken, r.Host, "oidcClaimsCacheKey unavailable")
```

Comment:
oidcClaimsCacheKey as string should use the defined const (also line 101)

Reply:
makes sense!